admin/class-h5p-plugin-admin.php

```diff
@@ -156,6 +159,79 @@ function add_settings_link($links) {
     return $links;
   }
 
+  /**
+   * Spawn an h5p content via POST request
+   */
+  public function spawn_h5p() {
+    $plugin = H5P_Plugin::get_instance();
+    $action = $_REQUEST['action'];
+    if ($action) {
+      $core = $plugin->get_h5p_instance('core'); // Make sure core is loaded
+      if ($action === 'create_h5p') {
+        // Handle creation of new h5p content.
+        $json = file_get_contents('php://input');
+
+        $decoded = json_decode($json);
+        if(!isset($decoded->library) || !isset($decoded->params)){
```

Comment:
My PHP is old and rusty, but could you explain why you  have to parse JSON and can't just access these via `$_POST['library']` and `$_POST['params']` ?